File: Programmers/Lv4/Lv4_숫자블록.py
import math
def get_num(target):        # ?먯떊???쒖쇅??理쒕? ?쎌닔瑜?諛섑솚
    end = int(math.sqrt(target))
    if end*end<target:
        end += 1
    temp = 2

    for i in range(2,end+1):
        temp = i
        if target%i==0:
            break

    if temp * temp > target:
        return 1;
    return target//temp


def solution(begin, end):
    answer = []
    start_block = 1
    end_block = end>>1

    if begin==1:
        answer.append(0)
        begin += 1

    for i in range(begin,end+1):
        answer.append(get_num(i))
    return answer

print(solution(1, 10))

File: Programmers/Lv4/Lv4_숫자블록.cpp
#include <string>
#include <vector>

using namespace std;

long long get_num(long long target) {		// 자신을 제외한 최대 약수를 반환
	long long i;
	for (i = 2; i*i < target; i++) {
		if (target%i==0)
			break;
	}
	if (i*i > target)
		return 1;
	return target / i;
}

// 각 자리는 최대 약수가 들어감
vector<int> solution(long long begin, long long end) {
	vector<int> answer;
	long long start_block = 1;
	long long end_block = end >> 1;

	if (begin == 1) {
		answer.push_back(0);
		begin++;
	}

	for (long long i = begin; i <= end; i++) {
		answer.push_back(get_num(i));
	}

	return answer;
}
